src/scripts/main.js

```diff
@@ -1,3 +1,163 @@
 'use strict';
 
-// write your code here
+let matrix = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+const message = [
+  document.querySelector('.message-start'),
+  document.querySelector('.message-win'),
+  document.querySelector('.message-lose'),
+];
+const buttonStart = document.querySelector('.start');
+const gameFieldCells = document.querySelectorAll('.field-cell');
+const infoScore = document.querySelector('.game-score');
+let gameScore = +(infoScore.textContent);
+const ROW_LENGTH = 4;
+const PROBILITY_NUMBER = 0.1;
+
+addRandomCell();
+addRandomCell();
+
+buttonStart.addEventListener('click', () => {
+  if (buttonStart.classList.contains('start')) {
+    goStart();
+  } else {
+    goEnd();
+  }
+});
+
+function goStart() {
+  buttonStart.classList.remove('start');
+  buttonStart.classList.add('restart');
+  buttonStart.textContent = 'restart';
+
+  message[0].classList.add('hidden');
+  addRandomCell();
+};
+
+function goEnd() {
```

Comment:
Same issue as the previous one. Use a separate class for game status and toggle it instead of manually adding and removing classes 'start' and 'restart'.